src/QuadForm/Herm/Genus.jl

```diff
@@ -1528,15 +1563,21 @@ Given a global genus symbol `G` for hermitian lattices over $E/K$, return a herm
 lattice over $E/K$ which admits `G` as global genus symbol.
 """
 function representative(G::GenusHerm)
+  if !is_integral(G)
+    s = denominator(scale(G))
+    L = representative(rescale(G, s))
+    return rescale(L, 1//s)
+  end
   P = _non_norm_primes(G.LGS)
   E = base_field(G)
   V = hermitian_space(E, _hermitian_form_with_invariants(base_field(G), rank(G), P, G.signatures))
   @vprint :Lattice 1 "Finding maximal integral lattice\n"
   M = maximal_integral_lattice(V)
-  lp = G.primes
-  for g in G.LGS
-    p = prime(g)
+  lp = primes(G)
+  lp = union(lp, support(discriminant(maximal_order(base_field(G)))))
```

Comment:
Eh I guess because I removed the unimodular parts at the bad primes... I was getting bugs otherwise but couldn't figure out why, that would be the reason